lib/index.js

```diff
@@ -1,105 +1,92 @@
+    this.files = {};
+    this.base = base;
+    process.addListener('exit', () => this.cleanupAllSync());
   }
-};
 
-function track(options) {
-  return Promise.try(function () {
-    // set up parameters
-    var filename;
-    var cleanupTimeout = AN_HOUR;
-    if (_.isString(options)) {
-      filename = options;
-    } else { // treat as object
-      filename = options.filename;
-      cleanupTimeout = options.cleanupTimeout || AN_HOUR;
+  getFilesBeingTracked() {
+    return _.values(this.files);
+  }
+  get BASE_DIR() {
+    return this.base;
+  }
+  set BASE_DIR(dir) {
+    this.base = dir;
+  }
+  async track(filename, timeout = HOUR, inDirectory = true) {
+    if (_.isObject(filename)) {
+      timeout = filename.cleanupTimeout || timeout;
+      filename = filename.filename;
+      inDirectory = filename.inDirectory;
     }
-    // full directory and file name
-    var pathName = path.join(baseDirectory, filename);
-
-    // add to array if it does not exist
-    if (fileTracking.indexOf(pathName) === -1) {
-      fileTracking.push(pathName);
-      // configure automatic cleanup
-      setTimeout(function (pathname) {
-        fs.access(pathname, function (err) {
-          if (err) {
-            // do nothing
-          } else {
-            fs.unlink(pathname, function () {
-              // do nothing
-            });
-          }
-        });
-      }, cleanupTimeout, pathName);
+    if (inDirectory) {
+      filename = path.join(this.base, filename);
     }
-    return pathName;
-  });
-}
+    if (this.files[filename]) {
+      return filename;
+    }
+    this.files[filename] = true;
+    // configure automatic cleanup
+    setTimeout(async (file) => {
+      try {
+        await fs.unlinkAsync(file);
+      } catch (err) {
+      // do nothing
+      }
+    }, timeout, filename);
+    return filename;
+  }
 
-function cleanupFile(pathName) {
-  return fs.unlinkAsync(pathName)
-    .catch(function () {
-      // ignore if it does not exist
-    })
-    .finally(function () {
-      _.remove(fileTracking, function (file) {
-        return file === pathName;
-      });
-    });
-}
+  async cleanupFile(pathname) {
+    try {
+      await fs.unlinkAsync(pathname);
+    } catch (err) {
+    // do nothing
+    } finally {
+      _.unset(this.files, pathname);
+    }
+  }
 
-function cleanupFileSync(pathName) {
-  try {
-    fs.unlinkSync(pathName);
-  } catch (err) {
+  cleanupFileSync(pathname) {
+    try {
+      fs.unlinkSync(pathname);
+    } catch (err) {
     // do nothing
+    }
+    _.unset(this.files, pathname);
   }
-  _.remove(fileTracking, function (file) {
-    return file === pathName;
-  });
-}
 
-function cleanupAll() {
-  return Promise.all(fileTracking, fs.unlinkAsync)
-    .catch(function () {
-      // do nothing
-    })
-    .finally(function () {
-      fileTracking = [];
-    });
-}
+  async cleanupAll() {
+    try {
+      await Promise.map(_.values(this.files), fs.unlinkAsync);
```

Comment:
Shouldn't it be `Promise.all` since you are not doing anything with the array returned?